packages/experiments/src/components/Button/ActionButton.tsx

```diff
@@ -5,35 +5,37 @@ import { ButtonVariantsType } from './ButtonVariants.types';
 import { FontWeights } from '../../Styling';
 
 const baseTokens: IButtonComponent['tokens'] = (props, theme): IButtonTokenReturnType => {
-  const { palette } = theme;
+  const { semanticColors } = theme;
 
   return {
     backgroundColor: 'transparent',
     backgroundColorHovered: 'transparent',
     backgroundColorPressed: 'transparent',
     borderColor: 'transparent',
-    color: palette.neutralPrimary,
-    colorHovered: palette.themePrimary,
-    colorPressed: palette.black,
+    borderColorHovered: 'transparent',
+    borderColorPressed: 'transparent',
+    color: semanticColors.buttonText,
+    colorHovered: semanticColors.link,
+    colorPressed: semanticColors.buttonTextCheckedHovered,
     contentPadding: '0px 8px',
     height: '40px',
-    iconColor: palette.neutralPrimary,
-    iconColorHovered: palette.themePrimary,
-    iconColorPressed: palette.black,
+    iconColor: semanticColors.buttonText,
+    iconColorHovered: semanticColors.link,
+    iconColorPressed: semanticColors.buttonTextCheckedHovered,
```

Comment:
Do these have the same semantic vs. palette concerns that were changed back to palette elsewhere?

Reply:
@Vitalius1 could you please take a look here.

Reply:
I don't think these ones quite match the semantics either. I would expect to see these:
```
    iconColor: semanticColors.buttonText,
    iconColorHovered: semanticColors.buttonTextHovered,
    iconColorPressed: semanticColors.buttonTextPressed
```

or even have some slots for the button icon for all states. This is why we need a good pass through all the `semanticColors` and maybe add if something is missing in order to truly have semantic meaning in their usage and not just picking the randomly to match the palette color they map too.